modules/dashboards/library/Dashboards/Form/DashletForm.php

```diff
@@ -0,0 +1,129 @@
+
+        $result = $this->getDb()->select($select);
+
+        foreach ($result as $dashboard) {
+            $dashboards[$dashboard->id] = $dashboard->name;
+        }
+
+        return $dashboards;
+    }
+
+    /**
+     * Create a new dashboard and return the last insert Id
+     * @param string $name
+     * @return int $id
+     */
+    public function createDashboard($name)
+    {
+        $data = [
+            'name' => $name
+        ];
+
+        $db = $this->getDb();
+        $db->insert('dashboard', $data);
+
+        $id = $db->lastInsertId();
+
+        return $id;
+    }
+
+    /**
+     * Display the FormElement for creating a new dashboards and dashlets
+     */
+    public function newAction()
+    {
+        $this->setAction('dashboards/dashlets/new');
+
+        $this->addElement('textarea', 'url', [
+            'label' => 'Url',
+            'placeholder' => 'Enter Dashlet Url',
+            'required' => true,
+            'rows' => '3'
+        ]);
+
+        $this->addElement('text', 'name', [
+            'label' => 'Dashlet Name',
+            'placeholder' => 'Enter Dashlet Name',
+            'required' => true
+        ]);
+
+        $this->addElement('checkbox', 'new-dashboard', [
+            'label' => 'New Dashboard',
+            'class' => 'autosubmit',
+        ]);
+
+        if ($this->getElement('new-dashboard')->getValue() === 'y') {
+            $this->addElement('text', 'new-dashboard-name', [
+                'label' => 'Dashboard Name',
+                'placeholder' => 'New Dashboard Name',
+                'required' => true,
+            ]);
+        } else {
+            $this->addElement('select', 'dashboard', [
+                'label' => 'Dashboard',
+                'required' => true,
+                'options' => $this->fetchDashboards()
+            ]);
+        }
+
+        $this->addElement('submit', 'submit', [
+            'label' => 'Add To Dashboard'
+        ]);
+    }
+
+    protected function assemble()
+    {
+        $this->add($this->newAction());
+    }
+
+    protected function onSuccess()
+    {
+        if ($this->getValue('new-dashboard')) {
+            if ($this->getValue('new-dashboard-name') !== null) {
+                $values = [
+                    'dashboard_id' => $this->createDashboard($this->getValue('new-dashboard-name')),
+                    'name' => $this->getValue('name'),
+                    'url' => $this->getValue('url')
+                ];
+
+                $this->getDb()->insert('dashlet', $values);
+                Notification::success('Dashlet in new Dashboard created');
+            } else {
+                Notification::error('Dashboard Name failed!');
+            }
+        } else {
+            $data = [
+                'dashboard_id' => $this->getValue('dashboard'),
+                'name' => $this->getValue('name'),
+                'url' => $this->getValue('url'),
+            ];
+
+            $this->getDb()->insert('dashlet', $data);
```

Comment:
No need to cache the variable `$data`. Just pass the data directly to `insert()`.